### DISCUSSION TITLE
svelte: mutation: why is it complaining about OperationStore if I'm passing a GraphQLRequest?

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1470] [Q&A] [answered]
I am trying to finish the Typescript conversion and now I have run into these kinds of errors with `@urql/svelte` [mutations](https://formidable.com/open-source/urql/docs/basics/svelte/#sending-a-mutation):

![image](https://user-images.githubusercontent.com/41120635/111849301-89897800-890d-11eb-8459-24652f3341c6.png)

Is there something I'm doing wrong?

```ts
const myMutation = `
  mutation ($id: ID!, $input: myInput!) {
    myUpdate(id: $id, input: $input) {
      id
      name
    }
  }
`;

const updateMutation = mutation({ query: myMutation });
```

Why is it complaining about `OperationStore` if I'm passing a `GraphQLRequest`?

```
Argument of type '{ query: string; }' is not assignable to parameter of type 'GraphQLRequest<any, object> | OperationStore<any, object>'.
  Type '{ query: string; }' is missing the following properties from type 'OperationStore<any, object>': variables, context, stale, fetching, and 6 more.ts(2345)
```

Anyway, I hate Typescript with all my heart! :rofl:

**Comment by kitten** [comment, marked as answer]
That's because it's not a complete request, so if you pass `variables` TypeScript will get it. I think we should actually make that type looser

**Reply by frederikhors** [reply]
Nope. I tried. It's the same.

![image](https://user-images.githubusercontent.com/41120635/111851162-8a70d880-8912-11eb-8af5-51ef9dce0487.png)

**Reply by frederikhors** [reply]
What do you mean is the type is wrong?
